core/src/main/scala/org/apache/spark/storage/DiskBlockManager.scala

```diff
@@ -48,32 +56,74 @@ private[spark] class DiskBlockManager(conf: SparkConf, deleteFilesOnStop: Boolea
   // of subDirs(i) is protected by the lock of subDirs(i)
   private val subDirs = Array.fill(localDirs.length)(new Array[File](subDirsPerLocalDir))
 
+  private[spark] val badDirs = ArrayBuffer[File]()
```

Comment:
how about call blacklistedDirs to keep naming consistent

Reply:
done